airbyte-integrations/connectors/source-s3/unit_tests/test_jsonl_parser.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import os
+from pathlib import Path
+from typing import Any, Mapping
+
+from source_s3.source_files_abstract.formats.jsonl_parser import JsonlParser
+
+from .abstract_test_parser import AbstractTestParser
+
+SAMPLE_DIRECTORY = Path(__file__).resolve().parent.joinpath("sample_files/")
+
+
+class TestJsonlParser(AbstractTestParser):
+    @classmethod
+    def cases(cls) -> Mapping[str, Any]:
+        return {
+            "basic_normal_test": {
+                "AbstractFileParser": JsonlParser(format={"filetype": "jsonl"}),
+                "filepath": os.path.join(SAMPLE_DIRECTORY, "jsonl/test_file_1.jsonl"),
+                "num_records": 8,
+                "inferred_schema": {
+                    "id": "integer",
+                    "name": "string",
+                    "valid": "boolean",
+                    "code": "integer",
+                    "degrees": "number",
+                    "birthday": "string",
+                    "last_seen": "string",
+                },
+                "line_checks": {},
+                "fails": [],
+            },
+            "encoding_Big5": {
+                "AbstractFileParser": JsonlParser(format={"filetype": "jsonl"}),
+                "filepath": os.path.join(SAMPLE_DIRECTORY, "jsonl/test_file_2_enc_Big5.jsonl"),
+                "num_records": 8,
+                "inferred_schema": {"id": "integer", "name": "string", "valid": "boolean"},
+                "line_checks": {},
+                "fails": [],
+            },
+            "encoding_Arabic_(Windows 1256)": {
+                "AbstractFileParser": JsonlParser(format={"filetype": "jsonl"}),
+                "filepath": os.path.join(SAMPLE_DIRECTORY, "jsonl/test_file_3_enc_Arabic.jsonl"),
+                "num_records": 2,
+                "inferred_schema": {"id": "integer", "notes": "string", "valid": "boolean"},
+                "line_checks": {},
+                "fails": [],
+            },
+            "compression_gz": {
```

Comment:
This test is failing because `birthday` is parsed as a python datetime object and not a string. Could you please make sure the master schema passed is enforced and when no schema is passed the dates are transformed to string?

Reply:
@alafanechere  As per document, it would infer as timestamp. JSON strings of the kind “YYYY-MM-DD” and “YYYY-MM-DD hh:mm:ss” convert to timestamp[s],

https://arrow.apache.org/docs/python/json.html#automatic-type-inference

Reply:
done